lib/translator/common.h

```diff
@@ -70,15 +67,41 @@ void SetAllocPageCallbacks(uint64_t (*alloc_callback)(uint16_t),
 const char* ScsiOpcodeToString(scsi::OpCode opcode);
 
 template <typename T>
-bool ReadValue(absl::Span<const uint8_t> data, T& out) {
+class Span {
+ public:
+  Span() : ptr_(nullptr), len_(0) {}
+  Span(T* ptr, size_t len) : ptr_(ptr), len_(len) {}
+  template <size_t N>
+  Span(T (&a)[N]) : Span(a, N) {}
+  template <typename Y,
+            typename = std::enable_if<std::is_convertible<T*, Y*>::value>>
+  Span(const Span<Y>& ref) : Span(ref.data(), ref.size()) {}
```

Comment:
I don't think this is sound, as a derived class pointer is convertible to a base class but may have different size, in which case pointer arithmetic will give incorrect results.

Do we actually need this interface at all?

Reply:
Good point, being more explicit here is best. We need this interface because the compiler doesn't implicitly know how to copy construct from Span<T> => Span<const T> (at least on my machine). This is something that ABSL supports and our library uses. I've pushed an update to the enable_if condition which should alleviate concerns of type size differences, and still enable the intended use case

Reply:
> We need this interface because the compiler doesn't implicitly know how to copy construct from Span<T> => Span<const T>

Ah, of course.

> I've pushed an update to the enable_if condition which should alleviate concerns of type size differences, and still enable the intended use case

That looks better, thanks. It might be safer still to more closely duplicate [what abseil does](https://github.com/abseil/abseil-cpp/blob/12bc53e0318d80569270a5b26ccbc62b52022b89/absl/types/internal/span.h#L59-L68) here, but your approach is simpler and there's a lot to be said for that, in the absence of any known shortcomings.